dash/dash.py

```diff
@@ -431,7 +433,7 @@ def __init__(  # pylint: disable=too-many-statements
         _get_paths.CONFIG = self.config
         _pages.CONFIG = self.config
 
-        self.pages_folder = pages_folder
+        self.pages_folder = str(pages_folder)
         self.use_pages = True if pages_folder != "pages" else use_pages
```

Comment:
Not new in this PR but since you're fixing some related edge cases... currently if you call:
```py
app = Dash(pages_folder="blah", use_pages=False)
```
the custom `pages_folder` wins and enables pages even though you explicitly said `use_pages=False`. I'm not sure why you'd do that (maybe to disable pages temporarily for debugging?), but it feels backward to me. Maybe we can switch `use_pages` to have a default of `None`, then this logic here can become something like:
```py
self.use_pages = (pages_folder != "pages") if use_pages is None else use_pages
```
Or maybe even centralize all of this logic (after changing the default `use_pages=None`) in `_configs.py`, then these lines can just draw from `config`:
```py
self.pages_folder = config.pages_folder
self.use_pages = bool(config.pages_folder)
```

Reply:
ahh yep, agreed, would be good to squeeze that in. I'll get onto it!

Reply:
Just a heads up @alexcjohnson that this suggestion didn't end up making it in, perhaps could be worth creating an issue for it?